Frameworks/Foundation/NSMutableSet.mm

```diff
@@ -139,11 +139,21 @@ - (void)removeObject:(id)object {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)filterUsingPredicate:(NSPredicate*)predicate {
-    UNIMPLEMENTED();
+    if (predicate == NULL) {
+        return;
+    }
+    @synchronized(self) {
+        NSArray* objects = [self allObjects];
+
+        for (id obj in objects) {
+            if (![predicate evaluateWithObject:obj]) {
+                [self removeObject:obj];
```

Comment:
>self removeObject:obj]; [](start = 17, length = 23)

It is not legal to mutate a collection while enumerating it.
Please build up a second set of removals and then use `-minusSet:` with it.